### ISSUE TITLE
backward not working for LookupTable

### ISSUE THREAD

**Issue by emergix** [Element-Research/rnn#62]
I have an example of use of LookupTable with the backward returning empty tensors (not computed)
is this a bug ?
gradInputs :	
{
  1 : DoubleTensor - size: 2x5
  2 : 
    {
      1 : DoubleTensor - empty
      2 : DoubleTensor - empty
      3 : DoubleTensor - empty
    }
}

code : 
-----------------------------------------------------
require 'nn'
require 'rnn'
require 'cutorch'
require 'cunn'

batchSize = 2
rho = 3
embeddingSize = 4
dictionarySize = 10
nbfeatures=5

inputs, targets = {}, {}            --  inputs and outputs
for i = 1, nbfeatures do   
       local featureTensor=torch.Tensor(batchSize,1)
        for j=1,batchSize do
          featureTensor[j][1]=torch.random(1,dictionarySize)
        end
        table.insert(inputs, featureTensor)
end
for i = nbfeatures+1, rho+nbfeatures do   
       local measure=torch.Tensor(batchSize)
        for j=1,batchSize do
          measure[j]=torch.random(1,dictionarySize)
        end
        table.insert(inputs, measure)
end
for i = 1, rho do  
        local measure=torch.Tensor(batchSize)
        for j=1,batchSize do
          measure[j]=torch.random(1,dictionarySize)
        end
      table.insert(targets, measure)
end

premodel=nn.Sequential()
b1=nn.Sequential()
b1:add(nn.NarrowTable(1,nbfeatures))
b1:add(nn.JoinTable(2))   -- ->Tensor(batchSize X nbfeatures)
b2=nn.Sequential()
b2:add(nn.NarrowTable(nbfeatures+1,rho))
c=nn.ConcatTable()
premodel:add(c)
c:add(b1)
c:add(b2)  -- ->{tensorF , {list of tensor(i)}}

inputsA=premodel:forward(inputs)
print('inputsA')
print(inputsA)

model=nn.Sequential()
p=nn.ParallelTable() 
p:add(nn.Identity())
p:add(nn.Sequencer(nn.LookupTable(dictionarySize, embeddingSize))) -- ->ListofTensor(batchSize X embeddingSize)
model:add(p)
SliceList=nn.ConcatTable() -- purpose: create a list tensor created by joining   tensorF & tensor(i)
for i=1, rho do   
    local Slice =nn.Sequential()
    SliceList:add(Slice)
    local cc=nn.ConcatTable()   -- contains the 2 tensors to join
    Slice:add(cc)
    local a=nn.Sequential()
    cc:add(a)
    a:add(nn.SelectTable(2)) -- we select list of tensor(i)
    a:add(nn.SelectTable(i))  -- we select a tensor(i)
    local b=nn.Sequential()
    cc:add(b)
    b:add(nn.SelectTable(1)) -- we select  tensorF
    Slice:add(nn.JoinTable(2)) -- we create a single tensor = tensorF & tensor(i)
end
model:add(SliceList)
model:add(nn.Sequencer(nn.FastLSTM(embeddingSize+nbfeatures, embeddingSize, rho)))
model:add(nn.Sequencer(nn.Linear(embeddingSize, dictionarySize)))
model:add(nn.Sequencer(nn.LogSoftMax()))

criterion = nn.SequencerCriterion(nn.ClassNLLCriterion())

prediction = model:forward(inputsA)
err = criterion:forward(prediction, targets)
print('err=' .. err)
gradOutputs = criterion:backward(prediction, targets)
gradInputs=model:backward(inputsA, gradOutputs)
print('gradInputs')
print(gradInputs)

**Comment by emergix** [comment]
I discovered that my problems come from the LookupTable.
this model cannot compute backward gradient, doing the backward phases fail.
The solution is to make this module the first to be executed in the model.

**Comment by nicholas-leonard** [comment]
@emergix yes, it's one of those annoying things we need to know about the LookupTable :)
